presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/FileResourceGroupConfigurationManager.java

```diff
@@ -0,0 +1,242 @@
+            for (ResourceGroupSpec candidate : candidates) {
+                if (candidate.getName().expandTemplate(session).equals(segments.get(i))) {
+                    if (i == segments.size() - 1) {
+                        checkState(match == null, "Ambiguous configuration for %s", group.getId());
+                        match = candidate;
+                    }
+                    else {
+                        checkState(nextCandidates == null, "Ambiguous configuration for %s", group.getId());
+                        nextCandidates = candidate.getSubGroups();
+                    }
+                }
+            }
+            if (nextCandidates == null) {
+                break;
+            }
+            candidates = nextCandidates;
+        }
+        checkState(match != null, "No matching configuration found for: %s", group.getId());
+        group.setSoftMemoryLimit(match.getSoftMemoryLimit());
+        group.setMaxQueuedQueries(match.getMaxQueued());
+        group.setMaxRunningQueries(match.getMaxRunning());
+    }
+
+    @Override
+    public List<? extends ResourceGroupSelector> get()
+    {
+        return selectors;
+    }
+
+    public static class ManagerSpec
+    {
+        private final List<ResourceGroupSpec> rootGroups;
+        private final List<SelectorSpec> selectors;
+
+        @JsonCreator
+        public ManagerSpec(
+                @JsonProperty("rootGroups") List<ResourceGroupSpec> rootGroups,
+                @JsonProperty("selectors") List<SelectorSpec> selectors)
+        {
+            this.rootGroups = ImmutableList.copyOf(requireNonNull(rootGroups, "rootGroups is null"));
+            this.selectors = ImmutableList.copyOf(requireNonNull(selectors, "selectors is null"));
+            Set<ResourceGroupNameTemplate> names = new HashSet<>();
+            for (ResourceGroupSpec group : rootGroups) {
+                checkArgument(!names.contains(group.getName()), "Duplicated root group: %s", group.getName());
+                names.add(group.getName());
+            }
+        }
+
+        public List<ResourceGroupSpec> getRootGroups()
+        {
+            return rootGroups;
+        }
+
+        public List<SelectorSpec> getSelectors()
+        {
+            return selectors;
+        }
+    }
+
+    public static class ResourceGroupSpec
+    {
+        private final ResourceGroupNameTemplate name;
+        private final DataSize softMemoryLimit;
+        private final int maxQueued;
+        private final int maxRunning;
+        private final List<ResourceGroupSpec> subGroups;
+
+        @JsonCreator
+        public ResourceGroupSpec(
+                @JsonProperty("name") ResourceGroupNameTemplate name,
+                @JsonProperty("softMemoryLimit") DataSize softMemoryLimit,
+                @JsonProperty("maxQueued") int maxQueued,
+                @JsonProperty("maxRunning") int maxRunning,
+                @JsonProperty("subGroups") List<ResourceGroupSpec> subGroups)
```

Comment:
I'd make this `Optional<List<ResourceGroupSpec>>` so you don't have to put `subGroups []` in the json